src/SmartFormat/Extensions/ListFormatter.cs

```diff
@@ -241,15 +241,15 @@ public bool TryEvaluateFormat(IFormattingInfo formattingInfo)
                 }
                 else if (CollectionIndex < items.Count - 1)
                 {
-                    spacerFormattingInfo.Write(spacer);
+                    spacerFormattingInfo.FormatAsChild(spacer, item);
                 }
                 else if (CollectionIndex == 1)
                 {
-                    spacerFormattingInfo.Write(twoSpacer);
+                    spacerFormattingInfo.FormatAsChild(twoSpacer, item);
```

Comment:
Only calling `FormatAsChild` if required, would be more performant

```Charp
private static void WriteSpacer(IFormattingInfo formattingInfo, Format? spacer, object? value)
{
    // Don't write the spacer
    if (spacer == null) return;

    if (spacer.HasNested)
    {
        formattingInfo.FormatAsChild(spacer, value);
    }
    else
    {
        formattingInfo.Write(spacer.GetLiteralText());
    }
}

```

Reply:
Good idea!

Reply:
I think we can skip the null check as I cant see when it would be null. We early out if `parameters.Count < 2 `